Framework/PythonInterface/test/python/plugins/algorithms/CreateGroupingByComponentTest.py

```diff
@@ -0,0 +1,124 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2025 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+from mantid.simpleapi import (
+    AnalysisDataService as ADS,
+    AlgorithmManager,
+)
+from testhelpers import run_algorithm
+import numpy as np
+
+
+class CreateGroupingByComponentTest(unittest.TestCase):
+    def setUp(self):
+        self.default_kwargs = {"InstrumentName": "ENGINX", "OutputWorkspace": "test_group"}
+
+    def tearDown(self):
+        ADS.clear()
+
+    def test_alg_with_default_parameters(self):
+        alg = run_algorithm("CreateGroupingByComponent", **self.default_kwargs)
+        self.assertTrue(alg.isExecuted())
+
+        # by default, include is "" which all components satisfy, so there should be 1 unique group
+        out_ws = ADS.retrieve(alg.getPropertyValue("OutputWorkspace"))
+        self.assertEqual(len(np.unique(out_ws.extractY())), 1)
+
+    def test_alg_with_no_search_term_splits_into_n_groups(self):
+        alg = run_algorithm("CreateGroupingByComponent", **self.default_kwargs, GroupSubdivision=5)
+        self.assertTrue(alg.isExecuted())
+
+        # by default, include is "" which all components satisfy, so sub-dividing this into 5 gives 5 unique groups
+        out_ws = ADS.retrieve(alg.getPropertyValue("OutputWorkspace"))
+        self.assertEqual(len(np.unique(out_ws.extractY())), 5)
+
+    def test_alg_with_include_term_splits_into_expected_groups(self):
+        alg = run_algorithm("CreateGroupingByComponent", **self.default_kwargs, IncludeComponents="detector-block")
+        self.assertTrue(alg.isExecuted())
+
+        # ENGINX has structure banks -> modules -> blocks -> pixels
+        # with 2 banks, each containing 5 modules (each module then made up of 9 blocks)
+
+        # here we have requested that we group the detector-blocks together.
+        # As this is done under each parent component we should have 10 groups + 1 for the null group
+
+        out_ws = ADS.retrieve(alg.getPropertyValue("OutputWorkspace"))
+        self.assertEqual(len(np.unique(out_ws.extractY())), 11)
```

Comment:
Is there a way to test the results (across all tests, not just this one) more robustly than just the number of entries in the workspace?